modules/oadp-configuring-aws-bsl.adoc

```diff
@@ -0,0 +1,70 @@
+----
+apiVersion: oadp.openshift.io/v1alpha1
+kind: BackupStorageLocation
+metadata:
+  name: default
+  namespace: openshift-adp
+spec:
+  provider: aws # <1>
+  objectStorage:
+    bucket: <bucket_name> # <2>
+    prefix: <bucket_prefix> # <3>
+  credential: # <4>
+    key: cloud # <5>
+    name: cloud-credentials # <6>
+  config:
+    region: <bucket_region> # <7>
+    s3ForcePathStyle: "true" # <8>
+    s3Url: <s3_url> # <9>
+    publicUrl: <public_s3_url> # <10>
+    serverSideEncryption: AES256 # <11>
+    kmsKeyId: "50..c-4da1-419f-a16e-ei...49f" # <12>    
+    customerKeyEncryptionFile: "/credentials/customer-key" # <13>
+    signatureVersion: "1" # <14>
+    profile: "default" # <15>
+    insecureSkipTLSVerify: "true" # <16>
+    enableSharedConfig: "true" # <17>
+    tagging: "" # <18>
+    checksumAlgorithm: "CRC32" # <19>
+----
+<1> The name of the object store plugin. In this example, the plugin is `aws`. This field is required.
+<2> The name of the bucket in which to store backups. This field is required.
+<3> The prefix within the bucket in which to store backups. This field is optional.
+<4> The credentials for the backup storage location. You can set custom credentials. If custom credentials are not set, the default credentials' secret is used.
+<5> The `key` within the secret credentials' data.
+<6> The name of the secret containing the credentials.
+<7> The AWS region where the bucket is located. Optional if s3ForcePathStyle is false.
+<8> A boolean flag to decide whether to use path-style addressing instead of virtual hosted bucket addressing. Set to "true" if using a storage service such as MinIO or NooBaa. This is an optional field. The default value is `false`.
+<9> You can specify the {aws-short} S3 URL here for explicitness. This field is primarily for storage services such as MinIO or NooBaa. This is an optional field.
+<10> This field is primarily used for storage services such as MinIO or NooBaa. This is an optional field.
+<11> The name of the server-side encryption algorithm to use for uploading objects, for example, `AES256`. This is an optional field.
+<12> Specify an {aws-short} KMS key ID. You can format, as shown in the example, as an alias, such as `alias/<KMS-key-alias-name>`, or the full `ARN` to enable encryption of the backups stored in S3. Note that, `kmsKeyId` cannot be used in with `customerKeyEncryptionFile`. This is an optional field.
+<13> Specify the file that has the `SSE-C` customer key to enable customer key encryption of the backups stored in S3. The file must contain a 32-byte string. The `customerKeyEncryptionFile` points to a mounted secret within the `velero` container. Add the following key-value pair to the `velero` `cloud-credentials` secret: `customer-key: <your_b64_encoded_32byte_string>`. Note that the `customerKeyEncryptionFile` field cannot be used with the `kmsKeyId`. The default value is an empty string (""), which means, `SSE-C` is disabled. This is an optional field.
```

Comment:
same point about backticks if it shows up in a file:
```suggestion
<13> Specify the file that has the `SSE-C` customer key to enable customer key encryption of the backups stored in S3. The file must contain a 32-byte string. The `customerKeyEncryptionFile` points to a mounted secret within the `velero` container. Add the following key-value pair to the `velero` `cloud-credentials` secret: `customer-key: <your_b64_encoded_32byte_string>`. Note that the `customerKeyEncryptionFile` field cannot be used with the `kmsKeyId`. The default value is an empty string (`""`), which means, `SSE-C` is disabled. This is an optional field.
```

Reply:
Done. thanks.